cpp/core/jni/JniWrapper.cc

```diff
@@ -431,95 +431,50 @@ JNIEXPORT jobject JNICALL Java_io_glutenproject_vectorized_ColumnarBatchOutItera
   auto executionCtx = jniCastOrThrow<ExecutionCtx>(ctxHandle);
 
   auto iter = executionCtx->getResultIterator(iterHandle);
-  std::shared_ptr<Metrics> metrics = iter->getMetrics();
-
-  int numMetrics = 0;
+  auto metrics = iter->getMetrics();
```

Comment:
Maybe we need to check and throw if `metrics` is nullptr, because it is still directly used as below.
> env->SetLongArrayRegion(longArray[i], 0, numMetrics, metrics->get((Metrics::TYPE)i));